sdk/core/azure-core/src/resource_identifier.cpp

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "azure/core/resource_identifier.hpp"
+
+#include <stdexcept>
+#include <string>
+
+namespace Azure { namespace Core {
+
+  ResourceIdentifier::ResourceIdentifier(std::string const& resourceId) : m_resourceId(resourceId)
+  {
+    const std::string subscriptionStart = "/subscriptions/";
+    const std::string providerStart = "/providers/";
```

Comment:
I think it is better to move these into `namespace {}`, just as is - no need to mark static (but PascalCasing them). Then, each time a constructor is called, it won't be creating two string objects.